myhoard/controller.py

```diff
@@ -1087,10 +1104,17 @@ def _get_upload_backup_site(self):
         return site_id, non_recovery_sites[site_id]
 
     def _get_site_for_stream_id(self, stream_id):
+        backup = self._get_backup_by_stream_id(stream_id)
+        if not backup:
+            KeyError(f"Stream {stream_id} not found in backups")
+        return backup["site"]
+
+    def _get_backup_by_stream_id(self, stream_id: str):
         for backup in self.state["backups"]:
             if backup["stream_id"] == stream_id:
-                return backup["site"]
-        raise KeyError(f"Stream {stream_id} not found in backups")
+                return backup
```

Comment:
We do filter backups by stream_id lots of times... but that sounds as a major change and multiple places on the code will be affected, not sure if it will be worth it at the end of the day.

Reply:
Okay, out of scope then. 👍 
At the end of the review, I'll create a few tickets about all the points mentioned, to keep track of them and they can be prioritized by Amichai at a later date.

Reply:
https://github.com/Aiven-Open/myhoard/pull/175 handles this